kratos/containers/sparse_graph.h

```diff
@@ -0,0 +1,394 @@
+        }
+        //set it to zero in parallel to allow first touching
+        #pragma omp parallel for
+        for(int i=0; i<static_cast<int>(nrows+1); ++i)
+            rRowIndices[i] = 0;
+
+        //count the entries TODO: do the loop in parallel if possible
+        for(const auto& item : this->GetGraph())
+        {
+            rRowIndices[item.first+1] = item.second.size();
+        }
+        //sum entries
+        for(int i = 1; i<static_cast<int>(rRowIndices.size()); ++i){
+            rRowIndices[i] += rRowIndices[i-1];
+        }
+
+
+        IndexType nnz = rRowIndices[nrows];
+        if(rColIndices.size() != nnz){
+            rColIndices.resize(nnz, false);
+        }
+        //set it to zero in parallel to allow first touching
+        #pragma omp parallel for
+        for(int i=0; i<static_cast<int>(rColIndices.size());++i)
+            rColIndices[i] = 0;
+
+        //count the entries TODO: do the loop in parallel if possible
+        for(const auto& item : this->GetGraph()){
+            IndexType start = rRowIndices[item.first];
+
+            IndexType counter = 0;
+            for(auto index : item.second){
+                rColIndices[start+counter] = index;
+                counter++;
+            }
+        }
+
+        //reorder columns
+        #pragma omp parallel for
+        for(int i=0; i<static_cast<int>(rRowIndices.size()-1);++i){
+            std::sort(rColIndices.begin()+rRowIndices[i], rColIndices.begin()+rRowIndices[i+1]);
+        }
+        return nrows;
+    }
+
+    ///@}
+    ///@name Operations
+    ///@{
+
+
+    ///@}
+    ///@name Access
+    ///@{
+
+
+    ///@}
+    ///@name Inquiry
+    ///@{
+
+
+    ///@}
+    ///@name Input and output
+    ///@{
+
+    /// Turn back information as a string.
+    virtual std::string Info() const
+    {
+        std::stringstream buffer;
+        buffer << "SparseGraph" ;
+        return buffer.str();
+    }
+
+    /// Print information about this object.
+    virtual void PrintInfo(std::ostream& rOStream) const {rOStream << "SparseGraph";}
+
+    /// Print object's data.
+    virtual void PrintData(std::ostream& rOStream) const {}
+
+    ///@}
+    ///@name Friends
+    ///@{
+
+
+    ///@}
+
+protected:
+    ///@name Protected static Member Variables
+    ///@{
+
+
+    ///@}
+    ///@name Protected member Variables
+    ///@{
+
+
+    ///@}
+    ///@name Protected Operators
+    ///@{
+
+
+    ///@}
+    ///@name Protected Operations
+    ///@{
+
+
+    ///@}
+    ///@name Protected  Access
+    ///@{
+
+
+    ///@}
+    ///@name Protected Inquiry
+    ///@{
+
+
+    ///@}
+    ///@name Protected LifeCycle
+    ///@{
+
+
+    ///@}
+
+private:
+    ///@name Static Member Variables
+    ///@{
+
+
+    ///@}
+    ///@name Member Variables
+    ///@{
+    GraphType mGraph;
+
+    ///@}
+    ///@name Private Operators
+    ///@{
+    friend class Serializer;
+
+    void save(Serializer& rSerializer) const
+    {
+        std::vector< IndexType > IJ;
```

Comment:
why you are copying it to a vector? You may store the size here and then in the load just make a loop with size and load the entities.

Reply:
essentially as this way i know that it is saved directly. I can change, however this is on the critical path, are u sure there are no performance implications?